contrib/twitchtv/twirp/twirp.go

```diff
@@ -0,0 +1,134 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package twirp
+
+import (
+	"context"
+	"math"
+	"net/http"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/twitchtv/twirp"
+)
+
+type contextKey int
+
+const (
+	twirpErrorKey contextKey = 0
+	httpSpanKey   contextKey = 1
+)
+
+// WrapServer wraps a http.Handler to add distributed tracing to a Twirp server.
+func WrapServer(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		opts := []tracer.StartSpanOption{
+			tracer.SpanType(ext.SpanTypeHTTP),
+			tracer.Tag(ext.HTTPMethod, r.Method),
+			tracer.Tag(ext.HTTPURL, r.URL.String()),
```

Comment:
All of the integrations I see use `r.URL.Path` for the `ext.HTTPURL`.
(https://github.com/DataDog/dd-trace-go/blob/v1/contrib/net/http/roundtripper.go#L32)

Reply:
We should be careful here. Ideally we should not set this tag.

* `r.URL.Path` contains only the path, which may have auth tokens or the like
* `r.URL.String()` contains the query string too, which again can have sensitive information.

Reply:
This was updated.